model/src/main/kotlin/config/OrtConfiguration.kt

```diff
@@ -174,6 +176,7 @@ data class OrtConfiguration(
                 .addEnvironmentSource()
                 .addPropertySources(sources)
                 .withContextResolverMode(ContextResolverMode.SkipUnresolved)
+                .withExplicitSealedTypes()
```

Comment:
I think the `ScanStorageConfiguration` is the only sealed class we use in the config, but I don't think that we use discriminator fields there. I'm not sure if this is properly tested.